src/node/plugins/localSearchPlugin.ts

```diff
@@ -97,7 +97,10 @@ export async function localSearchPlugin(
   function getDocId(file: string) {
     let relFile = path.relative(siteConfig.srcDir, file)
     relFile = siteConfig.rewrites.map[relFile] || relFile
-    let id = path.join(siteConfig.userConfig.base ?? '/', relFile)
+    let id = path.join(
+      siteConfig.userConfig.base ?? siteConfig.site.base,
```

Comment:
ah, just `siteConfig.site.base` should be sufficient, no need to use userConfig.base